src/template_compiler_utils.erl

```diff
@@ -56,8 +56,14 @@ next_context_var(CState, #ws{nr=Nr} = Ws) ->
 
 %% @doc Convert a list or binary to an atom.
 -spec to_atom(string()|binary()|atom()) -> atom().
-to_atom(L) when is_list(L) -> erlang:list_to_atom(L);
-to_atom(B) when is_binary(B) -> erlang:binary_to_atom(B, 'utf8');
+to_atom(L) when is_list(L) -> case catch erlang:list_to_existing_atom(L) of
```

Comment:
It no need to add this logic here by reason that if atom already exists in the system, it will not create or re create new one, simple check:
```erlang
1> erlang:memory(atom_used).
256891
2> list_to_atom("zotonic").
zotonic
3> erlang:memory(atom_used).
258010
4> list_to_atom("zotonic"). 
zotonic
5> erlang:memory(atom_used).
258010
6> list_to_existing_atom("zotonic").
zotonic
7> erlang:memory(atom_used).        
258010
```
As you can see, the size of atoms is not growing.

Reply:
Thanks for the informative link.
Yes, atom leak can be a problem, but I was wrong in thinking that `list_to_atom` or `binary_to_atom` always increases memory usage.